lib/freetype/index.js

```diff
@@ -247,16 +252,30 @@ module.exports.glyph_render = function (face, code, opts = {}) {
   let advance_x = glyph_data.linearHoriAdvance;
   let advance_y = glyph_data.linearVertAdvance;
 
+  let pixel_mode = glyph_data.bitmap.pixel_mode;
+
   let output = [];
 
   for (let y = 0; y < g_h; y++) {
     let row_start = buffer + y * pitch;
     let line = [];
 
     for (let x = 0; x < g_w; x++) {
-      if (opts.mono) {
+      if (pixel_mode === m.FT_PIXEL_MODE_MONO) {
```

Comment:
https://github.com/lvgl/lv_font_conv/blob/master/lib/collect_font_data.js#L108

`mono` is destination param (no subpixels, bpp == 1 => mono OLED). Idea was to use speical mono hinter for such case. You changes use only glyph info and ignore desinaion device. I did not dived into details, but feel something could be missed.

Reply:
`opts.mono` is still used a bit up in the file: https://github.com/treitmayr/lv_font_conv/blob/otf_gray_tones/lib/freetype/index.js#L178-L180 to select the way freetype renders the glyph. The above modification just changes the way the pixel conversion detects the pixel format when filling the `output` 2D vector. Of course this condition should be equivalent with `opts.mono`, but I wanted to base the decision on the actual provided pixel format as to avoid another flag for RGBA data. Also the freetype documentation is not 100% clear if multi-tone rendering will always be presented as RGBA (or actually ABGR) or possibly only in 8-bit gray values, see [this section](https://www.freetype.org/freetype2/docs/reference/ft2-base_interface.html#ft_load_xxx) for `FT_LOAD_COLOR`. Furthermore if no color palette is found (despite the color flag being set in the font) the rendering will also result in 8-bit gray values. So the changed code should be able to take whatever the result of freetype's rendering is.